WeakAurasOptions/ConditionOptions.lua

```diff
@@ -313,6 +314,10 @@ local function addControlsForChange(args, order, data, conditionVariable, totalA
       conditions[i].changes[j].value = v;
       WeakAuras.ClearAndUpdateOptions(data.id)
     end
+    setValueReload = function(info, v)
+      setValue(info, v)
+      WeakAuras.FillOptions()
```

Comment:
So the way AceConfig typically works is that after a control has fired an OnValueChanged event, the settter of that option is called, and then the options are reopened.
See, https://github.com/WoWUIDev/Ace3/blob/5250653df38b943a99c3826636d13a807a2261ed/AceConfig-3.0/AceConfigDialog-3.0/AceConfigDialog-3.0.lua#L658, which in the part at https://github.com/WoWUIDev/Ace3/blob/5250653df38b943a99c3826636d13a807a2261ed/AceConfig-3.0/AceConfigDialog-3.0/AceConfigDialog-3.0.lua#L844 calls AceConfigDialog:Open.

We use that by in our setters rebuilding the options table, with all necessary changes.

For "range" controls though, the options are not opened automatically but only on "OnMouseUp". This is most likely to prevent the widgets from being rebuilt while dragging. [1]

WeakAuras.FillOptions does essentially the same as AceConfigDialog:Open, that's why it mostly works.
There's another common case where we need to open the options manually, that is when we use a confirmation dialog

[1] This change actually seems to work even with drag and drop, which I can't really explain. It should be rebuilding the options and thus changing the widget, so that shouldn't work.